src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts

```diff
@@ -212,33 +212,34 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 					this.cellModel.markdownCursorPosition = selection?.getPosition();
 				}
 			}
+			const selection = window.getSelection();
+			const range = selection.rangeCount > 0 ? window.getSelection().getRangeAt(0) : undefined;
```

Comment:
Can re-use the selection object for that middle case. Also, do we still need to check if selection is undefined like in the previous code?

Reply:
yup definitely. nice catch, thanks.